homeassistant/components/mochad.py

```diff
@@ -33,18 +37,33 @@
     })
 }, extra=vol.ALLOW_EXTRA)
 
+CONFIG_SCHEMA_MQTT = vol.Schema({
+    DOMAIN_MQTT: vol.Schema({
+        vol.Required(CONF_BROKER, default='localhost'): cv.string,
+        vol.Required(CONF_PORT, default=1883): cv.port,
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
 
 def setup(hass, config):
     """Set up the mochad component."""
     conf = config[DOMAIN]
     host = conf.get(CONF_HOST)
     port = conf.get(CONF_PORT)
+    conf_mqtt = config[DOMAIN_MQTT]
+    broker = conf_mqtt.get(CONF_BROKER)
+    mqtt_port = conf_mqtt.get(CONF_PORT)
+    username = conf_mqtt.get(CONF_USERNAME)
+    password = conf_mqtt.get(CONF_PASSWORD)
 
     from pymochad import exceptions
 
     global CONTROLLER
     try:
-        CONTROLLER = MochadCtrl(host, port)
+        CONTROLLER = MochadCtrl(host, port, broker, mqtt_port, username,
```

Comment:
Don't use a global variable to store this. Store it in hass.data.

Reply:
wrapped global controller into hass.data